internal/cli/cmd/login/login.go

```diff
@@ -177,6 +182,11 @@ func (o *loginOptions) run(ctx context.Context) error {
 		refreshToken = ""
 	}
 
+	// To retain the existing user if he's logging into the same server
+	if currentConfig.APIAddress == o.ServerAddress {
+		o.Config = currentConfig
+	}
+
```

Comment:
As noted [here](https://github.com/akuity/kargo/pull/2373/files#r1702128376), `o.Config` _already_ contains the current config, which means this block never has any effect.

```suggestion
```